packages/lesswrong/integrationTests/integrationTestSetup.ts

```diff
@@ -0,0 +1,119 @@
+import { setSqlClient, closeSqlClient, getSqlClientOrThrow } from '../lib/sql/sqlClient';
+import {
+  preparePgTables,
+  createTestingSqlClientFromTemplate,
+  dropTestingDatabases,
+} from '../lib/sql/tests/testingSqlClient';
+
+// Work around an incompatibility between Jest and iconv-lite (which is used
+// by mathjax).
+require('iconv-lite').encodingExists('UTF-8')
+require('encoding/node_modules/iconv-lite').encodingExists('UTF-8')
+
+let dbConnected = false;
+async function ensureDbConnection() {
+  if (dbConnected)
+    return;
+
+  try {
+    const connectionString = process.env.MONGO_URL as string; //Provided by @shelf/jest-mongodb
+    const client = new MongoClient(connectionString, {
+      // See https://mongodb.github.io/node-mongodb-native/3.6/api/MongoClient.html
+      // for various options that could be tuned here
+
+      // A deprecation warning says to use this option 
+      useUnifiedTopology: true,
+    });
+    await client.connect();
+    const db = client.db();
+    setDatabaseConnection(client, db);
+  } catch(err) {
+    // eslint-disable-next-line no-console
+    console.error("Failed to connect to mongodb:", err);
+    return;
+  }
+
+  dbConnected = true;
+}
+
+let pgConnected = false;
+const ensurePgConnection = async () => {
+  if (!pgConnected) {
+    try {
+      preparePgTables();
+      const client = await createTestingSqlClientFromTemplate("unittest_jest_template");
+      setSqlClient(client);
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error("Failed to connect to postgres:", err.message);
+      throw err;
+    }
+    pgConnected = true;
+  }
+}
+
+let setupRun = false;
+async function oneTimeSetup() {
+  if (setupRun) return;
+  setupRun = true;
+
+  // We need to require this here instead of importing at the top level to make sure that
+  // Jest can do its magic to make ESM mocks work which requires calls to `jest.mock` to
+  // be evaluated before any of the mocked modules are loaded by node
+  require('../server');
+
+  setServerSettingsCache({});
+  setPublicSettings({});
+
+  await Promise.all([
+    ensurePgConnection(),
+    ensureDbConnection(),
+  ]);
+
+  await runStartupFunctions();
+
+  // define executableSchema
+  createVoteableUnionType();
+  initGraphQL();
+}
+
+jest.setTimeout(20000);
+
+beforeAll(async () => {
+  chai.should();
+  chai.use(chaiAsPromised);
+
+  await oneTimeSetup();
+});
+
+afterEach(async () => {
+  await waitUntilCallbacksFinished();
+});
+
+afterAll(async () => {
+  await waitUntilCallbacksFinished();
+  await Promise.all([
+    closeDatabaseConnection(),
+    closeSqlClient(getSqlClientOrThrow()),
+  ]);
+
+  // Our approach to database cleanup is to just delete all the runs older than 1 day.
+  // This allows us to inspect the databases created during the last run if necessary
+  // for debugging whilst also making sure that we clean up after ourselves eventually
+  // (assuming that the tests are run again some day).
```

Comment:
It's effectively a no-op in CI because the entire database instance is destroyed after the run completes, but it's still necessary to cleanup when people run integration tests on their own machine (unless we want to do some fancy Docker stuff, and I think the answer is no).